cedar-policy-core/src/entities/json/entities.rs

```diff
@@ -212,6 +213,39 @@ impl<'e, 's, S: Schema> EntityJsonParser<'e, 's, S> {
         Ok(entities.into_iter())
     }
 
+    /// Parse a single entity from an in-memory JSON value
+    pub fn single_from_json_value(
+        &self,
+        value: serde_json::Value,
+    ) -> Result<Entity, EntitiesError> {
+        let ejson: EntityJson =
+            serde_json::from_value(value).map_err(JsonDeserializationError::from)?;
+        let entity = self.parse_ejson(ejson)?;
+        match self.schema {
+            Some(schema) => {
+                let checker = crate::entities::conformance::EntitySchemaConformanceChecker::new(
+                    schema,
+                    self.extensions,
+                );
+                checker.validate_entity(&entity)?;
```

Comment:
I think this API is a potential footgun, wonder if it can be reworked

Reply:
`validate_entity()`?

Reply:
Yes. It seems easy to forget to call that, or have a flow through a function that doesn't call it. Violates the parse-don't-validate guideline

Reply:
That suggests an alternate API would have a different type for validated and not-yet-validated entities.  The validated one would get the `Entity` name I presume.